hi-ml-histopathology/src/histopathology/preprocessing/create_panda_tiles_dataset.py

```diff
@@ -40,15 +56,18 @@ def select_tile(mask_tile: np.ndarray, occupancy_threshold: float) \
         raise ValueError("Tile occupancy threshold must be between 0 and 1")
     foreground_mask = mask_tile > 0
     occupancy = foreground_mask.mean(axis=(-2, -1))
-    return (occupancy > occupancy_threshold).squeeze(), occupancy.squeeze()
+    selected = occupancy > occupancy_threshold
+    return selected[:, 0], occupancy[:, 0]
```

Comment:
Because squeezing removes all dimensions when only one tile is to be selected, which raises an error later on.

```python
In [1]: import numpy as np

In [2]: selected = np.zeros((128, 1))

In [3]: selected[:, 0].shape
Out[3]: (128,)

In [4]: selected.squeeze().shape
Out[4]: (128,)

In [5]: selected = np.zeros((1, 1))

In [6]: selected[:, 0].shape
Out[6]: (1,)

In [7]: selected.squeeze().shape
Out[7]: ()
```

Reply:
fair enough